libs/wingc/src/jsify.rs

```diff
@@ -1242,125 +1229,147 @@ impl<'ast> Visit<'ast> for FieldReferenceVisitor<'_> {
-			self.diagnostics.push(Diagnostic {
-				level: DiagnosticLevel::Error,
-				message: format!(
-					"Unable to reference \"this.{}\" from inflight method \"{}\" because type {} is not capturable",
-					property.name, self.method_name.name, field_kind.type_,
-				),
-				span: Some(property.span.clone()),
-			});
+			// if the field is a resource, it means we have "this.my_bucket" without an operation
+			// so we are going to emit a diagnostic that says we cannot qualify the reference.
+			if variable.type_.as_resource().is_some() {
+				self.diagnostics.push(Diagnostic {
+					level: DiagnosticLevel::Error,
+					message: format!(
+						"Cannot qualify which operations are performed on resource \"this.{}\" (see https://github.com/winglang/wing/issues/76)",
+						property.name,
+					),
+					span: Some(property.span.clone()),
+				});
+			}
 
-			return true;
-		}
+			// if this is a preflight field, it must be non-reassignable
+			if variable.reassignable {
+				self.diagnostics.push(Diagnostic {
+					level: DiagnosticLevel::Error,
+					message: format!(
+						"Cannot reference a reassignable field \"{}\" (\"var\") from an inflight context",
+						property.name,
+					),
+					span: Some(property.span.clone()),
+				});
 
-		debug!(
-			"field \"this.{}\" is referenced from inflight method {}",
-			property.name, self.method_name.name
-		);
+				return false;
+			}
 
-		// get/create an "ops" set for this field
-		let ops = self.references.entry(property.name.clone()).or_default();
-
-		if !self.path.is_empty() {
-			let op = self.path[0].clone();
-
-			// if we are referencing a resource, verify that the operation is an inflight method
-			if let Some(r) = field_kind.type_.as_resource() {
-				if r
-					.get_method(&op)
-					.filter(|v| v.flight == Phase::Inflight && !v.is_static)
-					.is_none()
-				{
-					self.diagnostics.push(Diagnostic {
-						level: DiagnosticLevel::Error,
-						message: format!(
-							"Unable to reference \"{}\" from inflight method \"{}\" because it is not an inflight method",
-							self.path.join("."),
-							self.method_name.name,
-						),
-						span: Some(property.span.clone()),
-					});
-
-					return true;
-				}
+			// and of a capturable type
+			if !variable.type_.is_capturable() {
+				self.diagnostics.push(Diagnostic {
+					level: DiagnosticLevel::Error,
+					message: format!(
+						"Cannot reference non-capturable field \"{}\" from an inflight context",
+						property.name,
+					),
+					span: Some(property.span.clone()),
+				});
+
+				return false;
 			}
 
-			ops.insert(op);
-		}
+			// add the field to the list of references
+			self
+				.references
+				.entry(node.to_string())
+				.or_insert_with(|| BTreeSet::new());
 
-		self.path.clear();
+			return true;
+		} else {
+			// method call
+
+			// cannot call preflight methods (supposed to be covered during type checking)
+			assert!(variable.flight != Phase::Preflight);
+
+			// cannot call static methods
```

Comment:
should this be a TODO? or did we decide inflight static methods aren't a thing for now?

Reply:
I don't see a reason why this won't be a thing. I'll add a todo.